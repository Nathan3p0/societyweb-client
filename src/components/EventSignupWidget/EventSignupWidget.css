.eventsignup__container {
    background: #fff;
    margin: 2.2rem .5rem;
    box-shadow: 1px 3px 10px #33333363;
}

.success {
    color: rgb(0, 141, 151);
}

.eventsignup__container-header {
    background-color: #26495E;
    color: white;
    padding: .3rem;
    padding-left: 1rem;
    font-size: 1.5rem;
}

.eventsignup__container-content {
    padding: 1rem;
    font-size: 1.1rem;
}

.eventsignup__form h3 {
    margin: 1rem 0;
}

.eventsignup__form li {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    margin-top: .5rem
}

.eventsignup__form > li > label {
    flex: 1 0 100px;
    font-size: 1.5rem;
}

.eventsignup__form > li > label + * {
    flex: 1 0 220px;
}

.eventsignup__form li button {
    margin-left: auto;
    padding: 8px 16px;
    border: none;
    background: #333333;
    color: #f2f2f2;
    text-transform: uppercase;
    letter-spacing: .09em;
    border-radius: 2px;
}

.eventsignup__form li button:hover {
    cursor: pointer;
    transform: scale(1.1);
    box-shadow: 1px 1px 5px #333333;
}

.eventsignup__form li input {
    padding: .5rem;
    border: 1px solid #33333341;
}

.eventsignup__form li input:hover,
.eventsignup__form li label:hover {
    cursor: pointer;
}

@media all and (max-width: 600px) {
    .eventsignup__container {
        margin: 1rem 0;
    }
}