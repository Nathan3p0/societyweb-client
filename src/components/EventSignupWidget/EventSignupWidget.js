import React from 'react';
import './EventSignupWidget.css';

const EventSignupWidget = (props) => {
    const { event } = props;
    return (
        <section className="eventsignup__container">
            <h3 className="eventsignup__container-header">Event Signup:</h3>
            <div className="eventsignup__container-content">
                {event &&
                    <p><strong>Event Name:</strong> {event.event_name}</p>}
                {!event && <h4>Select an event to signup</h4>}
                {event &&
                    <>
                        <p><strong>Event Description:</strong> {event.event_description}</p>
                        <form onSubmit={props.handleSignupSubmit}>
                            <ul className="eventsignup__form">
                                <h3>Are you interested?</h3>
                                {props.error && <p className="error">{props.error}</p>}
                                <li>
                                    <input type="radio" id="yes" name="attending" value="Yes" onChange={props.handleInputChange} />
                                    <label htmlFor="yes">Yes?</label>
                                </li>
                                <li>
                                    <input type="radio" id="maybe" name="attending" value="Maybe" onChange={props.handleInputChange} />
                                    <label htmlFor="maybe">Maybe?</label>
                                </li>
                                <li>
                                    <input type="radio" id="no" name="attending" value="No" onChange={props.handleInputChange} />
                                    <label htmlFor="no">No?</label>
                                </li>
                                <li>
                                    <input type="radio" id="volunteer" name="attending" value="Volunteer" onChange={props.handleInputChange} />
                                    <label htmlFor="volunteer">Volunteer?</label>
                                </li>
                                <li>
                                    <button type="submit">Submit Signup</button>
                                </li>
                            </ul>
                        </form>
                    </>
                }
            </div>
        </section>
    );
}

export default EventSignupWidget;