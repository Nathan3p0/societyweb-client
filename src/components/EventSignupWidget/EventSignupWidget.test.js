import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import EventSignupWidget from './EventSignupWidget'

describe('Event Signup Widget Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<EventSignupWidget />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<EventSignupWidget />).toJSON();
        expect(tree).toMatchSnapshot()
    })
})