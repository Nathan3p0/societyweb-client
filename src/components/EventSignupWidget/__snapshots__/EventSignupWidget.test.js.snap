// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Signup Widget Component Rendering Tests renders to UI as expected 1`] = `
<section
  className="eventsignup__container"
>
  <h3
    className="eventsignup__container-header"
  >
    Event Signup:
  </h3>
  <div
    className="eventsignup__container-content"
  >
    <h4>
      Select an event to signup
    </h4>
  </div>
</section>
`;
